### PR TITLE
Add golangci-lint to CI

### PR DESCRIPTION
**Pull request by cep21**
We should also task to improve these lint failures until the linter passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by luluzhao** [comment]
@cep21 build is broken which said it cannot parse .travis.yaml file. Could you check the format to make sure it can be built?

File: .travis.yml [hunk 1]
```diff
@@ -41,10 +41,12 @@ install:
     else
       go get;
     fi
+  - [ $DEP_MANAGEMENT_TOOL == "DEP" ] || go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.16.0
```
Comment (luluzhao) [inline .travis.yml]: I am wondering what you want to achieve here. Do you mean if `$DEP_MANAGEMENT_TOOL == "DEP"` and then `go get ...`?

Comment (cep21) [reply .travis.yml]: I'm not sure if we need to run a linter in *every* version of go, or just the latest version of Go.  A lot of the linting tools are newer and only work on the latest version of Go.

I think it makes a lot of sense to run unit tests or integration tests on every version of Go, but unsure what advantage we get out of limiting linters to just those supported on 1.9.

**Comment by cep21** [comment]
> build is broken which said it cannot parse .travis.yaml file.

Ah!  Where did you find that link (I cannot find this build on https://travis-ci.org/aws/aws-xray-sdk-go/pull_requests)

File: .travis.yml (hunk 1 above)
Comment (cep21) [reply .travis.yml]: Oh I misunderstood your question!

I'm trying to install golangci-lint only if modules are enabled, which is only if we're using modules and not dep.

**Comment by luluzhao** [comment]
@cep21 it is inside the Requests and here is the link: https://travis-ci.org/aws/aws-xray-sdk-go/requests

File: .travis.yml (hunk 1 above)
Comment (luluzhao) [reply .travis.yml]: in this case, we can make if else condition(there are some example inside the .travis.yaml file)

**Comment by luluzhao** [comment]
@cep21 thanks for the update and it works now. I am wondering you are you going to address the result of running the golangci-lint?

**Comment by cep21** [comment]
There is a lot of output.  Some don't matter and we can take out of the linter (like math/rand instead of crypto/rand ), but some are things we should resolve.  I worry it would be too big a diff to do everything at once, but having the list at least in CI can give us a target to reach.  What are your thoughts?

**Comment by luluzhao** [comment]
@cep21 Yeah, I agree with you. A big diff can be too hard to read. Let's break down into several PRs to cover them. :)

**Pull request merged**